src/main/java/ch/njol/skript/expressions/ExprPotionEffectTier.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * This file is part of Skript.
+ *
+ * Skript is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Potion Effect Tier")
+@Description("Returns the tier of an entities potion effect.")
+@Examples("if the amplifier of haste of player >= 3:")
+@Since("INSERT VERSION")
+public class ExprPotionEffectTier extends SimpleExpression<Number> {
+
+	static {
+		Skript.registerExpression(ExprPotionEffectTier.class, Number.class, ExpressionType.SIMPLE,
+			"[the] [potion] (tier|amplifier|level) of %potioneffecttype% of %livingentity%");
```

Comment:
```suggestion
		Skript.registerExpression(ExprPotionEffectTier.class, Number.class, ExpressionType.COMBINED,
			"[the] [potion] (tier|amplifier|level) of %potioneffecttype% (of|for) %livingentity%"
		);
```
SIMPLE is only for plain text expressions :)

Also, if possible, I think it would be good if both were plural (e.g. %potioneffecttypes% and %livingentities%)

Reply:
I don't think adding plural would be a good idea, as the list returned would be gibberish because of the mixed results?

Reply:
Yeah true - keeping it single is fine :)